Xamarin.Forms.Platform.Android/Renderers/MediaElementRenderer.cs

```diff
@@ -0,0 +1,381 @@
+using Android.Content;
+using Android.Media;
+using Android.Views;
+using Android.Widget;
+using System;
+using Xamarin.Forms;
+using Xamarin.Forms.Platform.Android;
+using Android.Graphics;
+using Android.Runtime;
+
+
+
+namespace Xamarin.Forms.Platform.Android
+{
+    public sealed class MediaElementRenderer : ViewRenderer<MediaElement, FrameLayout>, MediaPlayer.IOnCompletionListener, MediaPlayer.IOnPreparedListener, IMediaElementRenderer
+    {
+        MediaController _controller;
+        FormsVideoView _view;
+        MediaPlayer _mediaPlayer;
+
+		public MediaElementRenderer(Context context) : base(context)
+		{
+			AutoPackage = false;
+		}
+
+		double IMediaElementRenderer.BufferingProgress
+        {
+            get
+            {
+                if (_view != null)
+                {
+                    return _view.BufferPercentage / 100;
+                }
+
+                return 0.0;
+            }
+        }
+
+        TimeSpan IMediaElementRenderer.NaturalDuration
+        {
+            get
+            {
+                if (_view != null)
+                {
+                    return _view.NaturalDuration;
+                }
+
+                return TimeSpan.Zero;
+            }
+        }
+
+        int IMediaElementRenderer.NaturalVideoHeight
+        {
+            get
+            {
+                if (_view != null)
+                {
+                    return _view.VideoHeight;
+                }
+
+                return 0;
+            }
+        }
+
+        int IMediaElementRenderer.NaturalVideoWidth
+        {
+            get
+            {
+                if (_view != null)
+                {
+                    return _view.VideoWidth;
+                }
+
+                return 0;
+            }
+        }
+
+        void IMediaElementRenderer.Seek(TimeSpan time)
+        {
+            if (Control != null)
+            {
+                try
+                {
+                    _view?.SeekTo((int)time.TotalMilliseconds);
+                }
+                catch (ObjectDisposedException)
+                {
+                }
+            }
+        }
+
+        TimeSpan IMediaElementRenderer.Position
+        {
+            get
+            {
+                try
+                {
+                    return TimeSpan.FromMilliseconds(_view.CurrentPosition);
+                }
+                catch
+                {
+                    return TimeSpan.Zero;
```

Comment:
Is there a reason to for using `TimeSpan.Zero` to represent error or inapplicable cases, and elsewhere in this PR? Wouldn't a `Nullable<TimeSpan>` be better?
For Position, a `TimeSpan.Zero` could indicate an error (e.g. no loaded video), being at the start of a video.
For NaturalDuration, a `TimeSpan.Zero` could indicate an error, or being in a streaming video with unknown length.

Reply:
It roughly follows the UWP control although that uses a Duration for the NaturalDuration property - which supports an Auto value when it is unknown. I'm updating to make this one nullable with a null value having the same purpose as Duration.Auto. I'll also add a CanSeek property which will return false if no video is loaded.